### ISSUE TITLE
homebridge-enphase-envoy v4.3.2-beta119 - TypeError: Cannot read property 'activeCount' of undefined

### ISSUE THREAD

**Issue by hoefs** [grzegorz914/homebridge-enphase-envoy#14]
```
[2/10/2021, 3:21:22 PM] [enphaseEnvoy] Device: 192.168.2.25 Envoy-S, state: Online.
[2/10/2021, 3:21:22 PM] [enphaseEnvoy] -------- Envoy-S --------
[2/10/2021, 3:21:22 PM] [enphaseEnvoy] Manufacturer: Enphase
[2/10/2021, 3:21:22 PM] [enphaseEnvoy] Model: Envoy-S
[2/10/2021, 3:21:22 PM] [enphaseEnvoy] Meters: 0
[2/10/2021, 3:21:22 PM] [enphaseEnvoy] Q-Relays: 1
[2/10/2021, 3:21:22 PM] [enphaseEnvoy] Encharges: 0
[2/10/2021, 3:21:22 PM] [enphaseEnvoy] Inverters: 14
[2/10/2021, 3:21:22 PM] [enphaseEnvoy] Firmware: R4.10.35
[2/10/2021, 3:21:22 PM] [enphaseEnvoy] SerialNr: 121921106049
[2/10/2021, 3:21:22 PM] [enphaseEnvoy] Time: 2/10/2021, 3:21:19 PM
[2/10/2021, 3:21:22 PM] [enphaseEnvoy] ----------------------------------
[2/10/2021, 3:21:27 PM] [enphaseEnvoy] Device: 192.168.2.25 Envoy-S, update Device state error: TypeError: Cannot read property 'activeCount' of undefined
    at envoyDevice.updateDeviceState (/homebridge/node_modules/homebridge-enphase-envoy/index.js:1363:69)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
```

**Comment by hoefs** [comment]
Seems it is always in the loop:

```
[2/10/2021, 3:26:36 PM] [enphaseEnvoy] Device: 192.168.2.25 Envoy-S, state: Online.
[2/10/2021, 3:26:36 PM] [enphaseEnvoy] Device 192.168.2.25 Envoy-S, get device status data inventory [ [Object], [Object], [Object] ] info: <?xml version='1.0' encoding='UTF-8'?>
<envoy_info>
  <time>1612967194</time>
  <device>
    <sn>121921106049</sn>
    <pn>800-00553-r02</pn>
    <software>R4.10.35</software>
    <euaid>4c8675</euaid>
    <seqnum>0</seqnum>
    <apiver>1</apiver>
    <imeter>false</imeter>
  </device>
  <package name='rootfs'>
    <pn>500-00001-r01</pn>
    <version>02.00.00</version>
    <build>945</build>
  </package>
  <package name='kernel'>
    <pn>500-00011-r01</pn>
    <version>04.00.00</version>
    <build>5bb754</build>
  </package>
  <package name='boot'>
    <pn>590-00018-r01</pn>
    <version>02.00.01</version>
    <build>426697</build>
  </package>
  <package name='app'>
    <pn>500-00002-r01</pn>
    <version>04.10.35</version>
    <build>6ed292</build>
  </package>
  <package name='devimg'>
    <pn>500-00005-r01</pn>
    <version>01.02.186</version>
    <build>d0d70f</build>
  </package>
  <package name='geo'>
    <pn>500-00008-r01</pn>
    <version>02.01.22</version>
    <build>06e201</build>
  </package>
  <package name='backbone'>
    <pn>500-00010-r01</pn>
    <version>04.10.25</version>
    <build>7b7de5</build>
  </package>
  <package name='meter'>
    <pn>500-00013-r01</pn>
    <version>03.02.07</version>
    <build>4c9d48</build>
  </package>
  <package name='agf'>
    <pn>500-00012-r01</pn>
    <version>02.02.00</version>
    <build>c00a8f</build>
  </package>
  <package name='security'>
    <pn>500-00016-r01</pn>
    <version>02.00.00</version>
    <build>54a6dc</build>
  </package>
  <build_info>
    <build_time_gmt>1542157882</build_time_gmt>
    <build_id>release-4.10.x-103-Nov-12-18-18:25:06</build_id>
  </build_info>
</envoy_info>

[2/10/2021, 3:26:36 PM] [enphaseEnvoy] Device: 192.168.2.25 Envoy-S, get Device info.xml successful: {
  "envoy_info": {
    "time": [
      "1612967194"
    ],
    "device": [
      {
        "sn": [
          "121921106049"
        ],
        "pn": [
          "800-00553-r02"
        ],
        "software": [
          "R4.10.35"
        ],
        "euaid": [
          "4c8675"
        ],
        "seqnum": [
          "0"
        ],
        "apiver": [
          "1"
        ],
        "imeter": [
          "false"
        ]
      }
    ],
    "package": [
      {
        "$": {
          "name": "rootfs"
        },
        "pn": [
          "500-00001-r01"
        ],
        "version": [
          "02.00.00"
        ],
        "build": [
          "945"
        ]
      },
      {
        "$": {
          "name": "kernel"
        },
        "pn": [
          "500-00011-r01"
        ],
        "version": [
          "04.00.00"
        ],
        "build": [
          "5bb754"
        ]
      },
      {
        "$": {
          "name": "boot"
        },
        "pn": [
          "590-00018-r01"
        ],
        "version": [
          "02.00.01"
        ],
        "build": [
          "426697"
        ]
      },
      {
        "$": {
          "name": "app"
        },
        "pn": [
          "500-00002-r01"
        ],
        "version": [
          "04.10.35"
        ],
        "build": [
          "6ed292"
        ]
      },
      {
        "$": {
          "name": "devimg"
        },
        "pn": [
          "500-00005-r01"
        ],
        "version": [
          "01.02.186"
        ],
        "build": [
          "d0d70f"
        ]
      },
      {
        "$": {
          "name": "geo"
        },
        "pn": [
          "500-00008-r01"
        ],
        "version": [
          "02.01.22"
        ],
        "build": [
          "06e201"
        ]
      },
      {
        "$": {
          "name": "backbone"
        },
        "pn": [
          "500-00010-r01"
        ],
        "version": [
          "04.10.25"
        ],
        "build": [
          "7b7de5"
        ]
      },
      {
        "$": {
          "name": "meter"
        },
        "pn": [
          "500-00013-r01"
        ],
        "version": [
          "03.02.07"
        ],
        "build": [
          "4c9d48"
        ]
      },
      {
        "$": {
          "name": "agf"
        },
        "pn": [
          "500-00012-r01"
        ],
        "version": [
          "02.02.00"
        ],
        "build": [
          "c00a8f"
        ]
      },
      {
        "$": {
          "name": "security"
        },
        "pn": [
          "500-00016-r01"
        ],
        "version": [
          "02.00.00"
        ],
        "build": [
          "54a6dc"
        ]
      }
    ],
    "build_info": [
      {
        "build_time_gmt": [
          "1542157882"
        ],
        "build_id": [
          "release-4.10.x-103-Nov-12-18-18:25:06"
        ]
      }
    ]
  }
}
[2/10/2021, 3:26:36 PM] [enphaseEnvoy] -------- Envoy-S --------
[2/10/2021, 3:26:36 PM] [enphaseEnvoy] Manufacturer: Enphase
[2/10/2021, 3:26:36 PM] [enphaseEnvoy] Model: Envoy-S
[2/10/2021, 3:26:36 PM] [enphaseEnvoy] Meters: 0
[2/10/2021, 3:26:36 PM] [enphaseEnvoy] Q-Relays: 1
[2/10/2021, 3:26:36 PM] [enphaseEnvoy] Encharges: 0
[2/10/2021, 3:26:36 PM] [enphaseEnvoy] Inverters: 14
[2/10/2021, 3:26:36 PM] [enphaseEnvoy] Firmware: R4.10.35
[2/10/2021, 3:26:36 PM] [enphaseEnvoy] SerialNr: 121921106049
[2/10/2021, 3:26:36 PM] [enphaseEnvoy] Time: 2/10/2021, 3:26:34 PM
[2/10/2021, 3:26:36 PM] [enphaseEnvoy] ----------------------------------
[2/10/2021, 3:26:42 PM] [enphaseEnvoy] Debug production: {
  status: 200,
  statusText: 'OK',
  headers: [Object],
  config: [Object],
  request: [ClientRequest],
  data: [Object]
} productionCT: {
  status: 200,
  statusText: 'OK',
  headers: [Object],
  config: [Object],
  request: [ClientRequest],
  data: [Object]
}, inventory: {
  status: 200,
  statusText: 'OK',
  headers: [Object],
  config: [Object],
  request: [ClientRequest],
  data: [Array]
}
[2/10/2021, 3:26:44 PM] [enphaseEnvoy] Debug pcuCommCheck: {
  '121932043894': 4,
  '121932016547': 4,
  '121932034199': 3,
  '121932033746': 4,
  '121932016549': 3,
  '121932039512': 3,
  '121932038960': 3,
  '121932043380': 4,
  '121932039459': 4,
  '121910052727': 3,
  '121932039006': 4,
  '121931158966': 3,
  '121932035604': 3,
  '121931157713': 4,
  '121932039565': 3
}
[2/10/2021, 3:26:44 PM] [enphaseEnvoy] Device: 192.168.2.25 Envoy-S, update Device state error: TypeError: Cannot read property 'activeCount' of undefined
    at envoyDevice.updateDeviceState (/homebridge/node_modules/homebridge-enphase-envoy/index.js:1363:69)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
[2/10/2021, 3:27:03 PM] [Verisure] info Alarm (Thuis): Getting current alarm state.
[2/10/2021, 3:27:03 PM] [Verisure] info VOORDEUR: Getting current sensor state.
[2/10/2021, 3:27:03 PM] [Verisure] info KEUKENRAAM: Getting current sensor state.
[2/10/2021, 3:27:03 PM] [Verisure] info GARAGEDEUR: Getting current sensor state.
[2/10/2021, 3:27:03 PM] [UPS] Nut is Polling for Description Unavailable with polling delay of 60 seconds.
[2/10/2021, 3:27:03 PM] [UPS] Nut checking connection to Nut Server for Description Unavailable.
[2/10/2021, 3:27:03 PM] [UPS] Nut connection to Nut Server Successful.
[2/10/2021, 3:27:03 PM] [UPS] Nut refreshing delay 0.1 for Description Unavailable
[2/10/2021, 3:27:04 PM] [enphaseEnvoy] Device: 192.168.2.25 Envoy-S, requesting config information.
[2/10/2021, 3:27:04 PM] [UPS] Nut now getting vars for Description Unavailable.
[2/10/2021, 3:27:04 PM] [UPS] Nut now updating vars for Description Unavailable.
[2/10/2021, 3:27:05 PM] [enphaseEnvoy] Device: 192.168.2.25 Envoy-S, state: Online.
[2/10/2021, 3:27:05 PM] [enphaseEnvoy] Device 192.168.2.25 Envoy-S, get device status data inventory [ [Object], [Object], [Object] ] info: <?xml version='1.0' encoding='UTF-8'?>
<envoy_info>
  <time>1612967222</time>
  <device>
    <sn>121921106049</sn>
    <pn>800-00553-r02</pn>
    <software>R4.10.35</software>
    <euaid>4c8675</euaid>
    <seqnum>0</seqnum>
    <apiver>1</apiver>
    <imeter>false</imeter>
  </device>
  <package name='rootfs'>
    <pn>500-00001-r01</pn>
    <version>02.00.00</version>
    <build>945</build>
  </package>
  <package name='kernel'>
    <pn>500-00011-r01</pn>
    <version>04.00.00</version>
    <build>5bb754</build>
  </package>
  <package name='boot'>
    <pn>590-00018-r01</pn>
    <version>02.00.01</version>
    <build>426697</build>
  </package>
  <package name='app'>
    <pn>500-00002-r01</pn>
    <version>04.10.35</version>
    <build>6ed292</build>
  </package>
  <package name='devimg'>
    <pn>500-00005-r01</pn>
    <version>01.02.186</version>
    <build>d0d70f</build>
  </package>
  <package name='geo'>
    <pn>500-00008-r01</pn>
    <version>02.01.22</version>
    <build>06e201</build>
  </package>
  <package name='backbone'>
    <pn>500-00010-r01</pn>
    <version>04.10.25</version>
    <build>7b7de5</build>
  </package>
  <package name='meter'>
    <pn>500-00013-r01</pn>
    <version>03.02.07</version>
    <build>4c9d48</build>
  </package>
  <package name='agf'>
    <pn>500-00012-r01</pn>
    <version>02.02.00</version>
    <build>c00a8f</build>
  </package>
  <package name='security'>
    <pn>500-00016-r01</pn>
    <version>02.00.00</version>
    <build>54a6dc</build>
  </package>
  <build_info>
    <build_time_gmt>1542157882</build_time_gmt>
    <build_id>release-4.10.x-103-Nov-12-18-18:25:06</build_id>
  </build_info>
</envoy_info>

[2/10/2021, 3:27:05 PM] [enphaseEnvoy] Device: 192.168.2.25 Envoy-S, get Device info.xml successful: {
  "envoy_info": {
    "time": [
      "1612967222"
    ],
    "device": [
      {
        "sn": [
          "121921106049"
        ],
        "pn": [
          "800-00553-r02"
        ],
        "software": [
          "R4.10.35"
        ],
        "euaid": [
          "4c8675"
        ],
        "seqnum": [
          "0"
        ],
        "apiver": [
          "1"
        ],
        "imeter": [
          "false"
        ]
      }
    ],
    "package": [
      {
        "$": {
          "name": "rootfs"
        },
        "pn": [
          "500-00001-r01"
        ],
        "version": [
          "02.00.00"
        ],
        "build": [
          "945"
        ]
      },
      {
        "$": {
          "name": "kernel"
        },
        "pn": [
          "500-00011-r01"
        ],
        "version": [
          "04.00.00"
        ],
        "build": [
          "5bb754"
        ]
      },
      {
        "$": {
          "name": "boot"
        },
        "pn": [
          "590-00018-r01"
        ],
        "version": [
          "02.00.01"
        ],
        "build": [
          "426697"
        ]
      },
      {
        "$": {
          "name": "app"
        },
        "pn": [
          "500-00002-r01"
        ],
        "version": [
          "04.10.35"
        ],
        "build": [
          "6ed292"
        ]
      },
      {
        "$": {
          "name": "devimg"
        },
        "pn": [
          "500-00005-r01"
        ],
        "version": [
          "01.02.186"
        ],
        "build": [
          "d0d70f"
        ]
      },
      {
        "$": {
          "name": "geo"
        },
        "pn": [
          "500-00008-r01"
        ],
        "version": [
          "02.01.22"
        ],
        "build": [
          "06e201"
        ]
      },
      {
        "$": {
          "name": "backbone"
        },
        "pn": [
          "500-00010-r01"
        ],
        "version": [
          "04.10.25"
        ],
        "build": [
          "7b7de5"
        ]
      },
      {
        "$": {
          "name": "meter"
        },
        "pn": [
          "500-00013-r01"
        ],
        "version": [
          "03.02.07"
        ],
        "build": [
          "4c9d48"
        ]
      },
      {
        "$": {
          "name": "agf"
        },
        "pn": [
          "500-00012-r01"
        ],
        "version": [
          "02.02.00"
        ],
        "build": [
          "c00a8f"
        ]
      },
      {
        "$": {
          "name": "security"
        },
        "pn": [
          "500-00016-r01"
        ],
        "version": [
          "02.00.00"
        ],
        "build": [
          "54a6dc"
        ]
      }
    ],
    "build_info": [
      {
        "build_time_gmt": [
          "1542157882"
        ],
        "build_id": [
          "release-4.10.x-103-Nov-12-18-18:25:06"
        ]
      }
    ]
  }
}
[2/10/2021, 3:27:05 PM] [enphaseEnvoy] -------- Envoy-S --------
[2/10/2021, 3:27:05 PM] [enphaseEnvoy] Manufacturer: Enphase
[2/10/2021, 3:27:05 PM] [enphaseEnvoy] Model: Envoy-S
[2/10/2021, 3:27:05 PM] [enphaseEnvoy] Meters: 0
[2/10/2021, 3:27:05 PM] [enphaseEnvoy] Q-Relays: 1
[2/10/2021, 3:27:05 PM] [enphaseEnvoy] Encharges: 0
[2/10/2021, 3:27:05 PM] [enphaseEnvoy] Inverters: 14
[2/10/2021, 3:27:05 PM] [enphaseEnvoy] Firmware: R4.10.35
[2/10/2021, 3:27:05 PM] [enphaseEnvoy] SerialNr: 121921106049
[2/10/2021, 3:27:05 PM] [enphaseEnvoy] Time: 2/10/2021, 3:27:02 PM
[2/10/2021, 3:27:05 PM] [enphaseEnvoy] ----------------------------------
[2/10/2021, 3:27:08 PM] [enphaseEnvoy] Debug production: {
  status: 200,
  statusText: 'OK',
  headers: [Object],
  config: [Object],
  request: [ClientRequest],
  data: [Object]
} productionCT: {
  status: 200,
  statusText: 'OK',
  headers: [Object],
  config: [Object],
  request: [ClientRequest],
  data: [Object]
}, inventory: {
  status: 200,
  statusText: 'OK',
  headers: [Object],
  config: [Object],
  request: [ClientRequest],
  data: [Array]
}
[2/10/2021, 3:27:12 PM] [enphaseEnvoy] Debug pcuCommCheck: {
  '121932043894': 4,
  '121932016547': 4,
  '121932034199': 3,
  '121932033746': 4,
  '121932016549': 3,
  '121932039512': 3,
  '121932038960': 3,
  '121932043380': 4,
  '121932039459': 4,
  '121910052727': 3,
  '121932039006': 4,
  '121931158966': 3,
  '121932035604': 3,
  '121931157713': 4,
  '121932039565': 3
}
[2/10/2021, 3:27:12 PM] [enphaseEnvoy] Device: 192.168.2.25 Envoy-S, update Device state error: TypeError: Cannot read property 'activeCount' of undefined
    at envoyDevice.updateDeviceState (/homebridge/node_modules/homebridge-enphase-envoy/index.js:1363:69)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
```

**Comment by grzegorz914** [comment]
Try release 4.4.0

**Comment by grzegorz914** [comment]
Can u get me output from:
http://envoy.local/production.json

**Comment by grzegorz914** [comment]
Looks like U have Envoy-S standard, this model does not have any meters inputs. Fixed.

**Comment by hoefs** [comment]
> Looks like U have Envoy-S standard, this model does not have any meters inputs. Fixed.

Indeed envoy-s...

Performed the upgrade but got another error now:
![D9634D64-1E97-41A4-B962-7986178A336D](https://user-images.githubusercontent.com/7440522/107565290-d0d06a80-6be3-11eb-8769-ef40be9b06ac.jpeg)

**Comment by grzegorz914** [comment]
Please create manually in 
/homebridge/enphaseEnvoy/productionPowerMax_192168225 
file and restart Homebridge, then get the feedback

**Comment by hoefs** [comment]
> Please create manually in
> /homebridge/enphaseEnvoy/productionPowerMax_192168225
> file and restart Homebridge, then get the feedback

Do I need to create I manually? I have never had to do it. I know how to do it, but it is not user friendly.

**Comment by grzegorz914** [comment]
On which devices U run Homebridge? RPI or

**Comment by hoefs** [comment]
> On which devices U run Homebridge? RPI or

Docker on synology 

I have created the file. Seems like another (almost same name was existing; I have created productionPowerMax_xxxxxxxx and powerProductionMax_xxxxxxx was already existing)

![1CE348B8-19B6-4138-A0D2-5AEB3E3E8A36](https://user-images.githubusercontent.com/7440522/107568884-efd0fb80-6be7-11eb-9092-31012a3e7c94.jpeg)

**Comment by grzegorz914** [comment]
OK. fixed in 4.4.4
